File: testmodule/src/main/java/com/sunxy/testmodule/MainActivity.java
package com.sunxy.testmodule;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import com.sunxiaoyu.utils.UtilsCore;
import com.sunxiaoyu.utils.core.PhotoConfig;
import com.sunxiaoyu.utils.core.model.ActivityResultInfo;

import io.reactivex.disposables.Disposable;
import io.reactivex.functions.Consumer;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void onBackBtnClick(View view){
        Intent intent = new Intent(this, SecondActivity.class);
        Disposable subscribe = UtilsCore.manager().startForResult(this, 12, intent)
                .subscribe(new Consumer<ActivityResultInfo>() {
                    @Override
                    public void accept(ActivityResultInfo activityResultInfo) throws Exception {
                        Log.v("sun--", activityResultInfo.toString());
                    }
                });
    }

    public void onSelectPhotoClick(View view){
        UtilsCore.manager().selectPicture(this, 100)
                .subscribe(new Consumer<ActivityResultInfo>() {
                    @Override
                    public void accept(ActivityResultInfo activityResultInfo) throws Exception {
                        Log.v("sun--", activityResultInfo.toString());
                        String path = activityResultInfo.getData().getStringExtra(PhotoConfig.RESULT_PHOTO_PATH);
                        Log.v("sun--", path);
                    }
                });
    }

    public void onTakeClick(View view){
        String savePath = getExternalCacheDir().getAbsolutePath() + "/q.jpg";
        UtilsCore.manager().takePicture(this, 100,savePath)
                .subscribe(new Consumer<ActivityResultInfo>() {
                    @Override
                    public void accept(ActivityResultInfo activityResultInfo) throws Exception {
                        Log.v("sun--", activityResultInfo.toString());
                        String path = activityResultInfo.getData().getStringExtra(PhotoConfig.RESULT_PHOTO_PATH);
                        Log.v("sun--", path);
                    }
                });
    }
}

File: app/src/main/java/com/sunxiaoyu/sxyphoto/MainActivity.kt
package com.sunxiaoyu.sxyphoto

import android.Manifest
import android.content.Intent
import android.graphics.BitmapFactory
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.os.Environment
import android.util.Log
import android.view.View
import android.widget.ImageView
import com.sunxiaoyu.utils.UtilsCore
import com.sunxiaoyu.utils.core.PhotoConfig
import com.sunxiaoyu.utils.core.utils.DialogUtils

class MainActivity : AppCompatActivity() {


    val imageView: ImageView by lazy { findViewById(R.id.imageView) as ImageView }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    // 拍照
    fun takePhoto(view: View) {
        DialogUtils.showEditDialog(this, "请输入标签名"){_,_ ->

        }
//        val savePath = "${Environment.getExternalStorageDirectory().absolutePath}/sunxyPhoto/${System.currentTimeMillis()}.jpg"
//        UtilsCore.manager()
//                .takePicture(this, 10088, true, savePath)
//                .subscribe {
//                    Log.v("sunxy", it.toString())
//                    if (it.success()) {
//                        val path = it.data.getStringExtra(PhotoConfig.RESULT_PHOTO_PATH)
//                        setImage(path)
//                    }
//                }
    }

    // 图库选照
    fun selectPhoto(view: View) {
        val savePath = "${Environment.getExternalStorageDirectory().absolutePath}/sunxyPhoto/${System.currentTimeMillis()}.jpg"
        UtilsCore.manager()
                .selectPicture(this, 10086, true, savePath)
                .subscribe {
                    Log.v("sunxy", it.toString())
                    if (it.success()) {
                        val path = it.data.getStringExtra(PhotoConfig.RESULT_PHOTO_PATH)
                        setImage(path)
                    }
                }
    }

    // 跳转界面
    fun startA(view: View) {
        val intent = Intent(this, SecondActivity::class.java)
        intent.putExtra("n", "------------------------------------------9999990909090")
        UtilsCore.manager().startForResult(this, 10022, intent)
                .subscribe {
                    Log.v("sunxy", it.toString())
                    if (it.success()) {
                        val path = it.data.getStringExtra("n")
                        Log.v("sunxy", "result : $path")
                    }
                }
    }

    // 申请权限
    fun apply(view: View){
        UtilsCore.manager().requestPermissions(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
                .subscribe {
                    Log.v("sunxy", it.toString())
                }
    }

    // 查看图片
    fun setImage(path: String) {
        imageView.setImageBitmap(BitmapFactory.decodeFile(path))
        imageView.setOnClickListener {
            UtilsCore.manager().seePhoto(this, path)
        }
    }
}
